cpp/neuralnet/nninputs.cpp

```diff
@@ -641,6 +641,40 @@ Board SymmetryHelpers::getSymBoard(const Board& board, int symmetry) {
   return symBoard;
 }
 
+void SymmetryHelpers::maskSymmetricDuplicativeLoc(const Board &board, bool *const isSymDupLoc) {
+  fill(isSymDupLoc, isSymDupLoc + Board::MAX_ARR_SIZE, false);
+  vector<int> symTypes;
+  symTypes.reserve(NNInputs::NUM_SYMMETRY_COMBINATIONS-1);
+
+  for (int symmetry = 1; symmetry < NNInputs::NUM_SYMMETRY_COMBINATIONS; symmetry++) {
+    bool isBoardSym = true;
+    for (int y = 0; y < board.y_size; y++) {
+      for (int x = 0; x < board.x_size; x++) {
+        Loc loc = getSymLoc(x, y, board,symmetry);
+        if (board.colors[loc] != board.colors[Location::getLoc(x, y, board.x_size)]) {
+          isBoardSym = false;
+          break;
+        };
+      }
+      if (!isBoardSym) break;
+    }
+    if (isBoardSym) symTypes.push_back(symmetry);
+  }
+
+  //The way we iterate is to achieve https://senseis.xmp.net/?PlayingTheFirstMoveInTheUpperRightCorner%2FDiscussion
+
+  for(int x = board.x_size-1; x >= 0; x--)
```

Comment:
Minor: Each line seems to have an inconsistent brace and spacing style. Style should be:
```
for(int x = board.x_size-1; x >= 0; x--) {
  for(int y = 0; y < board.y_size; y++) {
    for(int symType: symTypes) {
```
Same for various places elsewhere.
I should try to set up a code formatter that automates some of these things. :)

Reply:
Just changed all.